Generation/src/components/GenMerge.cpp

```diff
@@ -0,0 +1,64 @@
+
+#include "GenMerge.h"
+
+#include "datamodel/MCParticleCollection.h"
+#include "datamodel/GenVertexCollection.h"
+
+#include "GaudiKernel/IIncidentSvc.h"
+#include "GaudiKernel/Incident.h"
+#include "GaudiKernel/IEventProcessor.h"
+
+DECLARE_COMPONENT(GenMerge)
+
+GenMerge::GenMerge(const std::string& name, ISvcLocator* svcLoc):
+  GaudiAlgorithm(name, svcLoc) {
+
+  declareInput("signalGenParticles", m_part_InSignal);
+  declareInput("signalGenVertices", m_vert_InSignal);
+
+  declareInput("pileupGenParticles", m_part_InPileUp);
+  declareInput("pileupGenVertices", m_vert_InPileUp);
+
+  declareOutput("allGenParticles", m_part_Out);
+  declareOutput("allGenVertices", m_vert_Out);
+}
+
+StatusCode GenMerge::initialize() {
+  StatusCode sc = GaudiAlgorithm::initialize();
+  if (!sc.isSuccess()) return sc;
+  return sc;
+}
+
+StatusCode GenMerge::execute() {
+  auto collVSig = m_vert_InSignal.get();
+  auto collPSig =  m_part_InSignal.get();
+
+  auto collVPil = m_vert_InPileUp.get();
+  auto collPPil =  m_part_InPileUp.get();
+
+  fcc::MCParticleCollection* collPOut = new fcc::MCParticleCollection();
+  fcc::GenVertexCollection* collVOut = new fcc::GenVertexCollection();
+
+  for(const auto elem: *collVSig) {
+    collVOut->push_back(elem.clone());
```

Comment:
As we just discussed, you'll also need to update the relations of the vertices here. Otherwise, if the signal / pile-up collection is dropped this will lead to inconsistent relations.